test-case/test-socwatch.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+set -e
+
+##
+## Case Name: check-soc-power-status
+## Preconditions:
+##    Intel SoCwatch tool must be installed on the device
+## Description:
+##    Run the socwatch command to check SoC power status
+## Test steps:
+##    1. load socwatch kernel module
+##    2. run socwatch command with desired parameters
+##    3. check return value from socwatch
+##    4. collect logs
+##    5. check dmesg errors
+##    6. unload socwatch kernel module
+## Expect result:
+##    consistent power status across tests
+##    check kernel log and find no errors
+##
+
+TOPDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+# shellcheck source=case-lib/lib.sh
+source "${TOPDIR}"/case-lib/lib.sh
+
+OPT_NAME['l']='loop'     OPT_DESC['l']='loop count'
+OPT_HAS_ARG['l']=1       OPT_VAL['l']=1
+
+OPT_NAME['d']='duration' OPT_DESC['d']='duration time for socwatch to collect the data'
+OPT_HAS_ARG['d']=1       OPT_VAL['d']=20
+
+OPT_NAME['w']='wait'     OPT_DESC['w']='idle time before starting socwatch to collect the data'
+OPT_HAS_ARG['w']=1       OPT_VAL['w']=5
+
+# TODO: socwatch test after unloading audio module might be useful
+OPT_NAME['u']='unload-audio'  OPT_DESC['u']='unload audio modules for the test'
+OPT_HAS_ARG['u']=0            OPT_VAL['u']=0
+
+: "${SOCWATCH_PATH:=$HOME/socwatch}"
+
+# reference cmd: sudo ./socwatch -t 20 -s 5 -f cpu-cstate -f pkg-pwr -o fredtest5
+#SOCWATCH_CMD="./socwatch"
+SOCWATCH_FEATURE_PARAMS=( -f cpu-cstate -f pkg-pwr )
+
+func_opt_parse_option "$@"
+func_lib_check_sudo
+duration=${OPT_VAL['d']}
+wait_time=${OPT_VAL['w']}
+loop_count=${OPT_VAL['l']}
+
+check_socwatch_module_loaded()
+{
+    lsmod | grep -q socwatch || die "socwatch is not loaded"
+}
+
+socwatch_test_once()
+{
+    local i="$1"
+    dlogi "===== Loop($i/$loop_count) ====="
+
+    # set up checkpoint for each iteration
+    setup_kernel_check_point
+
+    # load socwatch module
+    sudo "$SOCWATCH_PATH"/drivers/insmod-socwatch
```

Comment:
works OK if already loaded?

Reply:
Good point. If the driver is already loaded, insmod-socwatch return error and the test case will exit. I will handle this case.

```
The  driver is already loaded!  Use /home/ubuntu/socwatch/drivers/rmmod-socwatch to remove it.
```

Reply:
I add -q option, if the module is already loaded, we can go ahead with the testing. SoCWatch module is already picky, no need to unload and load again.